opentimelineio/algorithms/filter.py

```diff
@@ -0,0 +1,285 @@
+#!/usr/bin/env python
+#
+# Copyright 2018 Pixar Animation Studios
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+"""Algorithms for filtering OTIO files.  """
+
+import copy
+import opentimelineio as otio
+
+
+def _is_in(thing, container):
+    return any(thing is item for item in container)
+
+
+def _isinstance_in(child, typelist):
+    return any(isinstance(child, t) for t in typelist)
+
+
+def filtered_items(
+    input_object,
+    unary_filter_fn,
+    types_to_prune=None,
+    types_to_filter=None,
+):
+    """Filter input_object with unary_filter_fn to make a new copy of object.
+
+    unary_filter_fn will be called with the objects from input_object passed in
+    as arguments one a a time, including the input_object itself, in a depth
+    first traversal order.
+
+    If unary function returns an object, that object will be inserted into the
+    hierarchy, replacing whichever object was passed in to it.  If it returns
```

Comment:
Maybe for clarity this could say `replacing whichever object was passed in to it in the output.` This way it's clear that this isn't touching the original.